pkg/models/datasource_cache.go

```diff
@@ -17,10 +18,18 @@ func (ds *DataSource) getTimeout() time.Duration {
 	timeout := 0
 	if ds.JsonData != nil {
 		timeout = ds.JsonData.Get("timeout").MustInt()
+		if timeout == 0 {
```

Comment:
Hm, is it okay if someone actually sets the timeout to be 0 though?

Reply:
If timeout is set to 0 or if timeout is not set at all it will be zero due to `MustInt()`. Changed it to be greater than or equal to zero. In general if someone configures it to zero the global setting will be used. Make sense?

Reply:
Yes, I was just wondering what would happen if someone wanted that to be zero (which is wrong from all angles). Then it would ignore the fact that it's set to zero and search for a string value which would be wrong.

> In general if someone configures it to zero the global setting will be used

This explains things :)